<launch>
  <arg name="input_sensor_topic" default="/points_raw"/>
  <arg name="input_odom_topic" default="/odom"/>
  <arg name="input_imu_topic" default="/imu"/>

  <arg name="base_frame_id" default="base_link"/>
  <arg name="min_scan_range" default="5.0"/>
  <arg name="max_scan_range" default="200.0"/>
  <arg name="min_add_scan_shift" default="1.0"/>
  <arg name="ndt_res" default="5.0"/>
  <arg name="trans_eps" default="0.01"/>
  <arg name="step_size" default="0.1"/>
  <arg name="max_iter" default="30"/>
  <arg name="omp_num_thread" default="3"/>

  <node pkg="tf" type="static_transform_publisher" name="velodyne_transform_publisher" args="1.2 0 2.0 0 0 0 /base_link /velodyne 10" />

  <node pkg="ndt_slam" type="ndt_slam_node" name="ndt_slam_node" output="screen">
    <param name="base_frame_id" value="$(arg base_frame_id)"/>
    <param name="min_scan_range" value="$(arg min_scan_range)"/>
    <param name="max_scan_range" value="$(arg max_scan_range)"/>
    <param name="min_add_scan_shift" value="$(arg min_add_scan_shift)"/>
    <param name="ndt_res" value="$(arg ndt_res)"/>
    <param name="trans_eps" value="$(arg trans_eps)"/>
    <param name="step_size" value="$(arg step_size)"/>
    <param name="max_iter" value="$(arg max_iter)"/>
    <param name="omp_num_thread" value="$(arg omp_num_thread)"/>
    <remap from="points_raw" to="$(arg input_sensor_topic)"/>
    <remap from="odom" to="$(arg input_odom_topic)"/>
    <remap from="imu" to="$(arg input_imu_topic)"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ndt_slam)/rviz/ndt_slam.rviz"/>
</launch>
